Sources/SwiftFusion/Inference/FactorsStorage.swift

```diff
@@ -21,11 +21,12 @@ import PenguinStructures
 extension ArrayStorage where Element: Factor {
   /// Returns the errors, at `x`, of the factors.
   func errors(at x: VariableAssignments) -> [Double] {
-    Element.Variables.withVariableBufferBaseUnsafePointers(x) { varsBufs in
-      map { f in
-        f.error(at: Element.Variables(varsBufs, indices: f.edges))
+    let x = Element.Variables.withBufferBaseAddresses(x) { varsBufs in
+      self.map { f in
+        f.error(at: Element.Variables(at: f.edges, in: varsBufs))
       }
     }
+  return x
```

Comment:
indentation a bit off. also you can just return the expression instead of naming it `x`.